packages/suite/src/components/suite/Labeling/components/Metadata/withEditable.tsx

```diff
@@ -31,8 +41,9 @@ export const withEditable = (WrappedComponent: React.FC) => ({
 }: Props) => {
     const theme = useTheme();
     const [touched, setTouched] = useState(false);
+    // value is used to mirror divRef.current.textContent so that its changes force react to render
+    const [value, setValue] = useState('');
```

Comment:
Is this really needed, can't we just watch `divRef.current.textContent` in useEffect? Not sure tho, maybe I'm just confused with Vue 3...

Reply:
Probably yes. I need to trigger rerender somehow and it can be done only by props or state changes. Basically I need to know if there is some value so that I know whether placeholder should be rendered or not. If placeholder was a simple string, I could set divRef.current.textContent to it's value. But placeholder is typically a component.